src/Compilers/CSharp/Portable/BoundTree/BoundDagEvaluation.cs

```diff
@@ -10,13 +10,22 @@ namespace Microsoft.CodeAnalysis.CSharp
 {
     partial class BoundDagEvaluation
     {
-        public override bool Equals([NotNullWhen(true)] object? obj) => obj is BoundDagEvaluation other && this.Equals(other);
-        public virtual bool Equals(BoundDagEvaluation other)
+        public sealed override bool Equals([NotNullWhen(true)] object? obj) => obj is BoundDagEvaluation other && this.Equals(other);
+        public bool Equals(BoundDagEvaluation other)
         {
             return this == other ||
-                this.Kind == other.Kind &&
-                this.Input.Equals(other.Input) &&
-                Symbol.Equals(this.Symbol, other.Symbol, TypeCompareKind.AllIgnoreOptions);
+                this.IsEquivalentTo(other) &&
```

Comment:
This breaks the hashcode, because `IsEquivalentTo` will return true for otherwise-unequal `Symbol`s. Either the override of `object.Equals` should be comparing symbols with all comparison options, or hashcodes shouldn't include the symbol.

Reply:
`GetHashCode` is in sync with `Equals`.

`IsEquivalaentTo` is extracted to ignore the input in some cases - it's not supposed to reflect the hashcode.

Reply:
> GetHashCode is in sync with Equals.

It is not, that is why I'm commenting here. `GetHashCode` is based on `this.Symbol.GetHashCode()`, while `IsEquivalentTo` is based on comparison with `TypeCompareKind.AllIgnoreOptions`.

Reply:
In that case, this is a pre-existing issue. 

https://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp/BoundTree/BoundDagEvaluation.cs,d68789e6da5496b5,references

Reply:
Hmm. Alright, we'll leave this for now then.